src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -145,12 +149,33 @@ private Query getTermQuery(com.yelp.nrtsearch.server.grpc.TermQuery termQuery, I
     throw new IllegalArgumentException(String.format(message, termQuery, fieldDef.getType()));
   }
 
+  private void validateTermQuery(
+      FieldDef fieldDef, com.yelp.nrtsearch.server.grpc.TermQuery termQuery) {
+    validateTermQueryIsSearchable(fieldDef);
+    if (((TermQueryable) fieldDef).getTermQueryType() != termQuery.getTermTypesCase()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "%s field does not support term type: %s",
+              fieldDef.getName(), termQuery.getTermTypesCase()));
+    }
+  }
+
+  private void validateTermQueryIsSearchable(FieldDef fieldDef) {
+    if (fieldDef instanceof IndexableFieldDef
+        && !(fieldDef instanceof IdFieldDef)
```

Comment:
Should IdFieldDefs always be searchable? Was failing tests since the registerFields schema in the test didn't have it as searchable.

Reply:
I have made IdFieldDefs always required to be searchable.